### ISSUE TITLE
Streams only read to the 2nd last line

### ISSUE THREAD

**Issue by nickma42** [offdev/csv#3]
```
$parser = new Parser(
    stream("header\nrow 1\nrow 2\n")
);

foreach ($parser as $index => $row) {
    echo($row['header']);
}
```

gives the following output:

```
row1
```

The iterator tests only pass because `other-samples.csv` has trailing new lines;

```
$parser = new Parser(
    stream("header\nrow 1\nrow 2\n\n\n")
);

foreach ($parser as $index => $row) {
    echo($row['header']);
}
```

gives the following output:

```
row1
row2
```

**Comment by nickma42** [comment]
Also `rewind()` doesn't reset the current index so future iterations after a rewind have the wrong keys returned.

**Comment by offdev** [comment]
Sorry for the really, really delayed answer, didn't see the open issues...

Anyway, I fixed this a while ago ([1.2.0](https://github.com/offdev/csv/compare/1.1.1...1.2.0)), noticed it myself. Didn't see the index not resetting, and iterating twice over the same parser results in the header being taken as a value too.

I have now fixed your mentioned issues too ([1.2.1](https://github.com/offdev/csv/compare/1.2.0...1.2.1)), so I'll close this ticket.
